var/spack/repos/builtin/packages/qmcpack/package.py

```diff
@@ -0,0 +1,223 @@
+            args.append('-DMPI_BASE_DIR:PATH={0}'.format(mpi.prefix))
+
+        # Currently FFTW_HOME and LIBXML2_HOME are used by CMake.
+        # Any CMake warnings about other variables are benign.
+        xml2_prefix = self.spec['libxml2'].prefix
+        args.append('-DLIBXML2_HOME={0}'.format(xml2_prefix))
+        args.append('-DLibxml2_INCLUDE_DIRS={0}'.format(xml2_prefix.include))
+        args.append('-DLibxml2_LIBRARY_DIRS={0}'.format(xml2_prefix.lib))
+
+        fftw_prefix = self.spec['fftw'].prefix
+        args.append('-DFFTW_HOME={0}'.format(fftw_prefix))
+        args.append('-DFFTW_INCLUDE_DIRS={0}'.format(fftw_prefix.include))
+        args.append('-DFFTW_LIBRARY_DIRS={0}'.format(fftw_prefix.lib))
+
+        args.append('-DBOOST_ROOT={0}'.format(self.spec['boost'].prefix))
+        args.append('-DHDF5_ROOT={0}'.format(self.spec['hdf5'].prefix))
+
+        # Default is MPI, serial version is convenient for cases, e.g. laptops
+        if '+mpi' in self.spec:
+            args.append('-DQMC_MPI=1')
+        elif '~mpi' in self.spec:
+            args.append('-DQMC_MPI=0')
+
+        # Default is real-valued single particle orbitals
+        if '+complex' in self.spec:
+            args.append('-DQMC_COMPLEX=1')
+        elif '~complex' in self.spec:
+            args.append('-DQMC_COMPLEX=0')
+
+        # When '-DQMC_CUDA=1', CMake automatically sets:
+        # '-DQMC_MIXED_PRECISION=1'
+        #
+        # There is a double-precision CUDA path, but it is not as well
+        # tested. 
+
+        if '+cuda' in self.spec:
+            args.append('-DQMC_CUDA=1')
+        elif '~cuda' in self.spec:
+            args.append('-DQMC_CUDA=0')
+
+        # Mixed-precision versues double-precision CPU and GPU code        
+        if '+mixed' in self.spec:
+            args.append('-DQMC_MIXED_PRECISION=1')
+        elif '~mixed' in self.spec:
+            args.append('-DQMC_MIXED_PRECISION=0')
+
+        # New Structure-of-Array (SOA) code, much faster than default
+        # Array-of-Structure (AOS) code. 
+        # No support for local atomic orbital basis.
+        if '+soa' in self.spec:
+            args.append('-DENABLE_SOA=1')
+        elif '~soa' in self.spec:
+            args.append('-DENABLE_SOA=0')
+
+        # Manual Timers
+        if '+timers' in self.spec:
+            args.append('-DENABLE_TIMERS=1')
+        elif '~timers' in self.spec:
+            args.append('-DENABLE_TIMERS=0')
+        
+        # Include MKL flags     
+        if 'intel-mkl' in self.spec:
+            args.append('-DBLA_VENDOR=Intel10_64lp_seq')
+            args.append('-DQMC_INCLUDE={0}'.format(join_path(env['MKLROOT'],'include')))
+        return args
+
+    def setup_environment(self, spack_env, run_env):
+        # Add MKLROOT/lib to the CMAKE_PREFIX_PATH to enable CMake to find MKL libraries.
+        # MKLROOT environment variable must be defined for this to work properly.
+        if 'intel-mkl' in self.spec:
```

Comment:
if you set Blas/Lapack explicitly, you won'd need to bother with this part at all.

Reply:
The way CMAKE in QMCPACK is designed we use the built-in support for finding MKL. I believe this works with the Intel compiler without any extra effort. The complication comes when using non-Intel compilers with MKL. I will take a closer look at those references, thanks.

Reply:
> The way CMAKE in QMCPACK is designed we use the built-in support for finding MKL

by built-in, you probably refer to [find_package](https://cmake.org/cmake/help/v3.0/command/find_package.html), then you should be able to disable the whole search it by explicitly specifying blas/lapack.

Reply:
@davydden Here is the issue which I am not sure how to fix. My understanding is that with a non-Intel compiler, the MKL library *directory* must be included in the CMAKE_PREFIX_PATH -- otherwise, the built-in CMake (part of CMake, not part of QMCPACK) won't properly detect MKL.

Reply:
> the MKL library directory must be included in the CMAKE_PREFIX_PATH

it's up to how CMake is being used. But certainly not a requirement! I regularly compile with GCC+Intel-MKL at least these two package: [trilinos](https://github.com/LLNL/spack/blob/develop/var/spack/repos/builtin/packages/trilinos/package.py#L373-L378) and [dealii](https://github.com/LLNL/spack/blob/develop/var/spack/repos/builtin/packages/dealii/package.py#L185-L188) and everything works as expected.

Reply:
@naromero77 if you directly call `FIND_PACKAGE(LAPACK)`, then try following the way we configure `dealii` in spack: `<package>_FOUND` would disable any search and just provide the libs using Spack's interfaces to blas/lapack.